_inc/client/components/connect-button/index.jsx

```diff
@@ -8,6 +8,7 @@ import Button from 'components/button';
 import { translate as __ } from 'i18n-calypso';
 import analytics from 'lib/analytics';
 import getRedirectUrl from 'lib/jp-redirect';
+import UAParser from 'ua-parser-js';
```

Comment:
Instead of introducing a new dependency, I wonder if we could get that data from what we already have in Jetpack. Maybe we could rely on the Device Detection package that’s in Jetpack and pass that data to React here:
https://github.com/Automattic/jetpack/blob/5b9171eeea2ac30d0f5435b7e11d089960ef9999/_inc/lib/admin-pages/class.jetpack-react-page.php#L333
You can then fetch it in `_inc/client/state` and use it

Another alternative would be to rely on the global available in WP:
https://codex.wordpress.org/Global_Variables#Browser_Detection_Booleans

If we do choose to go with the new dependency, you'll want to explicitly declare it in `package.json`; we cannot just assume it will always be available to us as a dependency already there from existing packages.

Reply:
Many thanks @jeherve for your suggestions.
I have replaced the `ua-parser-js`  based functionality for checking `Safari` browser with an initial state reducer: `isSafari` and take advantage of the global `$is_safari` to pass the data to React.